src/node/rpc/frontend.h

```diff
@@ -454,45 +457,46 @@ namespace ccf
       update_history();
       update_consensus();
 
-#ifndef PBFT
       bool is_primary = (consensus == nullptr) || consensus->is_primary() ||
         ctx->is_create_request;
 
-      if (!is_primary)
+      if (consensus == nullptr || consensus->type() == ConsensusType::Raft)
```

Comment:
This doesn't look the like same condition we had before - why the `consensus == nullptr`? I would expect `ifndef PBFT` to become `consensus != nullptr && consensus->type() != ConsensusType::Pbft`?

Reply:
If you look at the line above, `bool is_primary = ...` this implies that when we were running in Raft at this point `consensus` could be `null`

Reply:
True, but this means that when running in Pbft and the consensus is null, we go into this branch where we didn't before. Since the next check is `!is_primary`, I think we can combine/simplify these checks.

Reply:
As discussed, throwing a logic error if `consensus == nullptr` since the check is most likely there for historical reasons. If we are processing the `create` transaction then we have already set the consensus. Will spend some time looking at it however.

Reply:
Actually this check should move up to `process()`